CNV-50248: Add a section with vSphere functionality mapping (#85687)
openshift/openshift-docs: modules/virt-vmware-comparison.adoc

// Module included in the following assemblies:
//
// * virt/about_virt/about-virt.adoc

:_mod-docs-content-type: REFERENCE
[id="virt-vmware-comparison_{context}"]
= Comparing {VirtProductName} to {vmw-full}

If you are familiar with {vmw-first}, the following table lists {VirtProductName} components that you can use to accomplish similar tasks. However, because {VirtProductName} is conceptually different from {vmw-short}, and much of its functionality comes from the underlying {product-title}, {VirtProductName} does not have direct alternatives for all {vmw-short} concepts or components.

.Mapping of {vmw-short} concepts to their closest {VirtProductName} counterparts
[options="header"]
[cols="2,2,3"]
|===
|{vmw-short} concept |{VirtProductName} |Explanation

|Datastore
|Persistent volume (PV){nbsp}+ +
Persistent volume claim (PVC)
|Stores VM disks. A PV represents existing storage and is attached to a VM through a PVC. When created with the `ReadWriteMany` (RWX) access mode, PVCs can be mounted by multiple VMs simultaneously.

|Dynamic Resource Scheduling (DRS)
|Pod eviction policy{nbsp}+ +

> Descheduler

Reviewer: Quick question about the table formatting:
are these extra `+` signs intentional?
![image](https://github.com/user-attachments/assets/6e9e88f3-34f7-46e6-a5ec-34bff772e893)

Because I think that just having the linebreaks like you do in the left column look better to me:
![image](https://github.com/user-attachments/assets/fef33538-ff4d-42be-bd88-16e0db7de85b)

Author: Hi @michaelryanpeter , the plus signs are intentional as both technologies in combination are a recommended alternative for either of the technologies listed in the first column. They are meant to be displayed and are followed by a line break to keep them cleanly on separate lines rather than breaking randomly.

Reviewer: Heard. Thanks for clarifying. It wasn't clear to me when I read it. I can go ahead and merge this if you are still ready?

|Provides active resource balancing. A combination of pod eviction policies and a descheduler allows VMs to be live migrated to more appropriate nodes to keep node resource utilization manageable.

|NSX
|Multus{nbsp}+ +
OVN-Kubernetes{nbsp}+ +
Third-party container network interface (CNI) plug-ins
|Provides an overlay network configuration. There is no direct equivalent for NSX in {VirtProductName}, but you can use the OVN-Kubernetes network provider or install certified third-party CNI plug-ins.

|Storage Policy Based Management (SPBM)
|Storage class
|Provides policy-based storage selection. Storage classes represent various storage types and describe storage capabilities, such as quality of service, backup policy, reclaim policy, and whether volume expansion is allowed. A PVC can request a specific storage class to satisfy application requirements.

|vCenter +

> vRealize Operations

Reviewer 2: I wonder if bullet points would improve clarity here (and in the vSwitch/DvSwitch cell)

// Module included in the following assemblies:
//
// * virt/about_virt/about-virt.adoc

:_mod-docs-content-type: CONCEPT

[id="virt-vmware-comparison_{context}"]
= Comparison of {VirtProductName} with VMware vSphere

> {VirtProductName} is conceptually different from VMware vSphere, much of its functionality comes from the underlying {product-title}, and it therefore does not offer direct alternatives to individual vSphere concepts and components. That said, if you are familiar with VMware vSphere, the following table lists the most common vSphere concepts and outlines what {VirtProductName} components you can use to accomplish a similar task.

Reviewer 2: I got a little carried away and rewrote this paragraph; please take what works and leave what doesn't. :) My main point here was to suggest that you start with what the table is for before getting into the differences/lack of direct alternatives.

```suggestion
If you are familiar with {vmw-first}, the following table might help you identify {VirtProductName} components that you can use to accomplish similar tasks. However, because {VirtProductName} is conceptually different from {vmw-short}, and much of its functionality comes from the underlying {product-title}, {VirtProductName} does not have direct alternatives for all {vmw-short} concepts or components.
```

.Mapping of vSphere concepts to their closest {VirtProductName} counterparts
[options="header"]
[cols="2,2,3"]
|===

> |vSphere concept |{VirtProductName} |Explanation

Reviewer 2: ```suggestion
|{vmw-short} concept |{VirtProductName} |Explanation
```
(applies throughout; just pointing this out as an example.)

|Datastore
|Persistent volume (PV){nbsp}+ +
Persistent volume claim (PVC)

> |Stores VM disks. A persistent volume represents existing storage and is attached to a VM through a persistent volume claim. When created with the `ReadWriteMany` (RWX) access mode, persistent volume claims can be mounted by multiple VMs simultaneously.

Reviewer 2: Because you defined the acronyms in the previous cell, you can use them here:
```suggestion
|Stores VM disks. A PV represents existing storage and is attached to a VM through a PVC. When created with the `ReadWriteMany` (RWX) access mode, PVCs can be mounted by multiple VMs simultaneously.
```

|Dynamic Resource Scheduling (DRS)
|Pod eviction policy{nbsp}+ +
Descheduler

> |Provides active resource balancing. A combination of pod eviction policies and a descheduler allows VMs to be live migrated to more appropriate nodes to keep node resource utilization manageable.

Reviewer 2: Suggestion to take or leave:
```suggestion
|Provides active resource balancing by allowing VMs to be live migrated to more appropriate nodes. This can keep node resource utilization manageable.
```

|NSX
|Multus{nbsp}+ +

> OVN-Kubernates{nbsp}+ +

Reviewer 2: Typo:
```suggestion
OVN-Kubernetes{nbsp}+ +
```

Third-party container network interface (CNI) plug-ins

> |Provides overlay network configuration. There is no direct equivalent for NSX in {VirtProductName}, depending on the desired functionality, users can either rely on the OVN-Kubernates network provider, or install certified third-party CNI plug-ins.

Reviewer 2: A few nits and questions:

* The [IBM Style Guide](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#d) says no to "desired":
> **desired** _adjective_
> Do not use. Use _required_ or _that you want_.
* Should there be an article (like "an") in "Provides overlay network configuration"?
* Typo in OVN-Kubernetes.
* The second sentence should be broken up into multiple sentences or shortened.

Suggestion to take or leave:
```suggestion
|Provides overlay network configuration. There is no direct equivalent for NSX in {VirtProductName}, but you can use the OVN-Kubernetes network provider or install certified third-party CNI plug-ins.
```

|Storage Policy Based Management (SPBM)
|Storage class

> |Provides policy-based storage selection. Storage classes represent various storage types and describe storage capabilities such as quality of service, backup policy, as well as the reclaim policy or whether volume expansion is allowed. Persistent volume claims can request a specific storage class to satisfy the application requirements.

Reviewer 2: This needs a little editing but is almost there. I don't think this suggestion is perfect, but perhaps heading in the right direction:

```suggestion
|Provides policy-based storage selection. Storage classes represent various storage types and describe storage capabilities, such as quality of service, backup policy, reclaim policy, and whether volume expansion is allowed. A PVC can request a specific storage class to satisfy application requirements.
```

|vCenter +
vRealize Operations
|OpenShift Metrics and Monitoring

> |Provides host and VM metrics. The {product-title} web console allows users to manage the VMs, view various metrics, and monitor the overall health of the cluster and VMs.

Reviewer 2: Applies in other places as well: please avoid using anthropomorphic words like "allows" when referring to components.

I'm also not sure that "manage the VMs" is relevant here, so I took it out in my suggestion:

```suggestion
|Provides host and VM metrics. You can view metrics and monitor the overall health of the cluster and VMs by using the {product-title} web console.
```

|vMotion
|Live migration

> |Allows the move of a running VM to another node without interruption. For live migration to be available, the persistent volume claim attached to the VM must have the `ReadWriteMany` (RWX) access mode.

Reviewer 2: ```suggestion
|Moves a running VM to another node without interruption. For live migration to be available, the PVC attached to the VM must have the `ReadWriteMany` (RWX) access mode.
```

|vSwitch +
DvSwitch
|NMState Operator{nbsp}+ +
Multus

> |Provides physical network configuration. The NMState Operator allows users to apply state-driven network configuration and manage various network interface types, including Linux bridges and network bonds. Multus allows users to attach multiple network interfaces and is often used to connect VMs to external networks.

Reviewer 2: Should there be an article in the first sentence, like "a" or "the"?

This is good overall, but please consider my previous comment re: "allows," and the general guidance to use active voice instead of passive voice wherever possible.
